File: scripts/generate-addon-info.json.py
#!/usr/bin/env python
"""Convert pyproject.toml to addon-info.json."""
import json
import sys
from argparse import Namespace

from translate_shell import __version__  # type: ignore

try:
    import tomllib  # type: ignore
except ImportError:
    import tomli as tomllib  # type: ignore


with open(sys.argv[1], "rb") as f:
    project = Namespace(**tomllib.load(f)["project"])

data = {
    "name": project.name,
    "description": project.description,
    "version": __version__.partition(".dev")[0],
    "author": project.authors[0]["name"]
    + " <"
    + project.authors[0]["email"]
    + ">",
    "repository": {"type": "git", "url": project.urls["Source"]},
}
print(json.dumps(data))

File: scripts/generate-addon-info.json.pl
#!/usr/bin/env -S perl -pi
use Env qw(GITHUB_REF_NAME);

BEGIN {
    unless ( $GITHUB_REF_NAME eq '' ) {
        $version = $GITHUB_REF_NAME;
    }
    else {
        $version = `git describe --abbrev=0`;
        chomp $version;
    }
}
unless ( $#ARGV < 0 ) {
    $name        = $1 if /^name = "([^"]*+)"/;
    $description = $1 if /^description = "([^"]*+)"/;
    $url         = $1 if /^Source = "([^"]*+)"/;
    $author      = "$1 <$2>"
      if /^authors = \[\{ name = "([^"]*+)", email = "([^"]*+)" }]/;
    next;
}
s/(?<="name": ")([^"]*+)(?=")/$name/;
s/(?<="description": ")([^"]*+)(?=")/$description/;
s/(?<="version": ")([^"]*+)(?=")/$version/;
s/(?<="author": ")([^"]*+)(?=")/$author/;
s/(?<="url": ")([^"]*+)(?=")/$url/;
